doc/mgmt/SONiC_DNS_Support.md

```diff
@@ -0,0 +1,407 @@
+
+# 2 Functionality
+
+## 2.1 Target Deployment Use Cases
+Manage/configure Management VRF via gNMI, REST and CLI interfaces
+
+## 2.2 Functional Description
+Provide CLI, gNMI and REST supports for Management VRF handling
+
+## 2.3 Backend change to support new configurations
+Provide change in hostcfgd, dns config script, dns service script.
+
+# 3 Design
+
+## 3.1 Overview
+
+Enhancing the management framework backend code and transformer methods to add support for DNS.
+
+## 3.2 DB Changes
+
+### 3.2.1 CONFIG DB
+```
+; DNS configuration attributes global to the system. Only one instance of the table exists in the system.
+; Key
+global_key           = “global”  ;  TACACS+ global configuration
+; Attributes
+src_ip               = IPAddress  ;  source IP address for the outgoing DNS packets
+
+```
+
+```
+; DNS name server configuration in the system.
+; Key
+server_key           = IPAddress  ;  DNS server’s address
+; Attributes
+No attributes are introduced as part of this design.
+```
+
+### 3.2.2 APP DB
+
+### 3.2.3 STATE DB
+
+### 3.2.4 ASIC DB
+
+### 3.2.5 COUNTER DB
+
+## 3.3 Switch State Service Design
+
+### 3.3.1 Orchestration Agent
+
+### 3.3.2 Other Process
+
+## 3.4 SyncD
+
+## 3.5 SAI
+
+## 3.6 User Interface
+
+### 3.6.1 Data Models
+
+YANG models needed for DNS handling in the management framework:
+1. **openconfig-system.yang**  
+
+2. **sonic-system-dns.yang**
+
+Supported yang objects and attributes:
+```diff
+
+module: openconfig-system
+      +--rw system
+
+      +--rw dns
+           |  +--rw config
+           |  |  +--rw search*                      oc-inet:domain-name
++          |  |  +--rw oc-sys-ext:source-address?   inet:ip-address
+           |  +--ro state
+           |  |  +--ro search*                      oc-inet:domain-name
++          |  |  +--ro oc-sys-ext:source-address?   inet:ip-address
+           |  +--rw servers
+           |  |  +--rw server* [address]
+           |  |     +--rw address    -> ../config/address
+           |  |     +--rw config
++          |  |     |  +--rw address?   oc-inet:ip-address
+           |  |     |  +--rw port?      oc-inet:port-number
+           |  |     +--ro state
++          |  |        +--ro address?   oc-inet:ip-address
+           |  |        +--ro port?      oc-inet:port-number
+
+Above fields with '+' are supported as part of DNS feature.
+
+module: sonic-system-dns
+  +--rw sonic-system-dns
+     +--rw DNS_SERVER
+     |  +--rw DNS_SERVER_LIST* [ipaddress]
+     |     +--rw ipaddress    inet:ip-address
+     +--rw DNS
+        +--rw DNS_LIST* [type]
+           +--rw type      enumeration ('global')
+           +--rw src_ip?   inet:ip-address
+
+```
+
+### 3.6.2 CLI
+
+
+#### 3.6.2.1 Configuration Commands
+All commands are executed in `configuration-view`:
+```
+sonic# configure terminal
+sonic(config)#
+```
+
+##### 3.6.2.1.1 Configure DNS server & source IP
+```
+sonic(config)# ip name-server
+  source-ip     Configure source IP to reach the name server

```

Comment:
Please rephrase this to "Configure Source IP, used for the DNS query"

Reply:
sure